sig/datadog/ci/contrib/patcher.rbs

```diff
@@ -0,0 +1,33 @@
+module Datadog
+  module CI
+    module Contrib
+      module Patcher
+        def self.included: (untyped base) -> untyped
+
+        module CommonMethods
+          attr_accessor patch_error_result: untyped
+
+          attr_accessor patch_successful: untyped
+
+          def patch_name: () -> String?
+
+          def name: () -> String
+
+          def patched?: () -> bool
+
+          def patch: () -> void
+
+          def on_patch_error: (untyped e) -> untyped
+
+          def default_tags: () -> untyped
```

Comment:
I know that it might be cumbersome to fill in, but maybe we can shape it a bit? Like if it's Array, we can state it?